src/frontends/tests/frontend/shared/test_builtin_extensions/builtin_extensions.cpp

```diff
@@ -2,12 +2,12 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-#include <openvino/core/extension.hpp>
-#include <openvino/frontend/extension/conversion.hpp>
-#include <openvino/opsets/opset8.hpp>
+#include "openvino/core/extension.hpp"
```

Comment:
why frontend build is disabled, if you build frontend tests? Looks strange

Reply:
The build options like:
```
        ENABLE_OV_PYTORCH_FRONTEND: OFF
        ENABLE_OV_PADDLE_FRONTEND: OFF
        ENABLE_OV_TF_FRONTEND: ON
        ENABLE_OV_TF_LITE_FRONTEND: ON
        ENABLE_OV_IR_FRONTEND: OFF
        ENABLE_OV_ONNX_FRONTEND: ON
```
Can be turned ON/OFF by user which remove some code from compilation. As result the other function are not used any more like  `CustomTranslatorTensorflow` which will cause warning as error for unused function. 
This PR is just disable unused code from building if is not used anywhere and will remove compilation warning.

The test code is build if `ENABLE_TESTS` is set to ON.